### PR TITLE
Support MSDK based decoding on Linux

### PR DESCRIPTION
**Pull request by taste1981**

### CODE REVIEW

File: talk/owt/sdk/include/cpp/owt/base/videorendererinterface.h [hunk 1]
```diff
@@ -70,6 +74,30 @@ class VideoRenderWindow {
   Window wnd_;
 };
 #endif
+#endif
+#if defined(WEBRTC_LINUX)
+typedef void (*PFN_RETURN_BUFFER)(void *data, uint32_t bufid);
+/// libva surface
+typedef void* VADisplay;        /* window system dependent */
+typedef unsigned int VASurfaceID;
+struct VaSurface {
```
Comment (jianjunz) [inline talk/owt/sdk/include/cpp/owt/base/videorendererinterface.h:83]: Descriptions are required for all public interfaces.

**Review by jianjunz — COMMENTED** [review]
If we plan to officially support Linux, perhaps we need to run CI on Linux as well.

File: build_overrides/webrtc.gni [hunk 2]
```diff
@@ -14,6 +14,9 @@ rtc_include_pulse_audio = true
 # Exclude internal ADM since Chromium uses its own IO handling.
 rtc_include_internal_audio_device = true
 
+# DO not use X11 for Linux
```
[the remark this answers is not in the document]
Comment (taste1981) [reply build_overrides/webrtc.gni:17]: done

File: scripts/build.sh [hunk 3]
```diff
@@ -0,0 +1 @@
+python build_linux.py --gn_gen --sdk --arch=x64 --msdk_root=/opt/intel/mediasdk/ --scheme=release
```
[the remark this answers is not in the document]
Comment (taste1981) [reply scripts/build.sh:1]: done.

File: scripts/build_linux.py [hunk 4]
```diff
@@ -29,6 +29,12 @@
     'is_component_build=false',
     'rtc_build_examples=false',
     'rtc_use_gtk=false',
+    'use_sysroot=false',
+    'is_clang=false',
+    'treat_warnings_as_errors=false',
+    'rtc_use_x11=false',
+    'rtc_include_pulse_audio=false',
+    'rtc_include_internal_audio_device=false'
```
[the remark this answers is not in the document]
Comment (taste1981) [reply scripts/build_linux.py:37]: Add option to disabling use internal audio device.

File: scripts/build_linux.py [hunk 5]
```diff
@@ -29,6 +29,12 @@
     'is_component_build=false',
     'rtc_build_examples=false',
     'rtc_use_gtk=false',
+    'use_sysroot=false',
```
[the remark this answers is not in the document]
Comment (taste1981) [reply scripts/build_linux.py:32]: done.

File: talk/owt/sdk/include/cpp/owt/base/videorendererinterface.h (hunk 1 above)
Comment (taste1981) [reply talk/owt/sdk/include/cpp/owt/base/videorendererinterface.h:83]: done.

File: talk/owt/sdk/include/cpp/owt/base/videorendererinterface.h [hunk 6]
```diff
@@ -9,8 +9,10 @@
 #include <Windows.h>
```
Comment (jianjunz) [inline talk/owt/sdk/include/cpp/owt/base/videorendererinterface.h:9]: Permission is modified from 644 to 755. It should be 644?

**Review by jianjunz — APPROVED** [review]
Only reviewed code outside talk/owt/sdk/base/linux/*. Code for MSDK seems to be similar as Windows version. I'm not quite familiar with it.

**Pull request merged**